projects/angular-calendar/src/modules/common/calendar-a11y.provider.ts

```diff
@@ -0,0 +1,154 @@
+import { Injectable } from '@angular/core';
+import { formatDate } from '@angular/common';
+import { A11yParams, CalendarA11yInterface } from './calendar-a11y.interface';
+import { pluralize } from './util';
+
+/**
+ * This class is responsible for adding accessibility to the calendar.
+ * You may override any of its methods via angulars DI to suit your requirements.
+ * For example:
+ *
+ * ```typescript
+ * import { A11yParams, CalendarA11y } from 'angular-calendar';
+ * import { formatDate } from '@angular/common';
+ *
+ * // adding your own a11y params
+ * export interface CustomA11yParams extends A11yParams {
+ *   isDrSuess?: boolean;
+ * }
+ *
+ * export class CustomCalendarA11y extends CalendarA11y {
+ *   // overriding a function
+ *   public openDayEventsLM({ date, locale, isDrSuess }: CustomA11yParams): string {
+ *     if (isDrSuess) {
+ *       return `
+ *         ${formatDate(date, 'EEEE MMMM d', locale)}
+ *          Today you are you! That is truer than true! There is no one alive
+ *          who is you-er than you!
+ *       `;
+ *     }
+ *   }
+ * }
+ *
+ * // in your component that uses the calendar
+ * providers: [{
+ *  provide: CalendarA11y,
+ *  useClass: CustomCalendarA11y
+ * }]
+ * ```
+ */
+@Injectable()
+export class CalendarA11y implements CalendarA11yInterface {
+  /**
+   * Aria label for the badges/date of a cell
+   * @example: `Saturday October 19 1 event`
+   */
+  public monthCell({ day, locale }: A11yParams): string {
+    if (day.badgeTotal > 0) {
+      return `
+        ${formatDate(day.date, 'EEEE MMMM d', locale)}
+         ${pluralize('event', day.badgeTotal, true)},
```

Comment:
Instead of this helper function, let's use the built in angular pipe for doing this: https://angular.io/api/common/I18nPluralPipe. So something like:
```
new I18nPluralPipe().transform(day.badgeTotal, {
  '=0': 'No events', '=1': 'One event', 'other': '# events'
})
```

Reply:
I had no idea this existed. Pretty cool. Obviously, the pluralize function was only made to handle English, so this is way better.

Reply:
@JaxonWright and I have tried using the I18nPluralPipe inside of the CalendarA11y provider without much success. @mattlewis92 can you provide any more help with how to implement this? Thanks 😄

Reply:
Ah, my bad, looks like it has a dependency which is why it's now working: https://github.com/angular/angular/blob/8.2.2/packages/common/src/pipes/i18n_plural_pipe.ts#L32

If you add it to the `providers` array in the `CalendarCommonModule` then you should be able to inject it into the constructor of the class using DI